community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java

```diff
@@ -206,14 +216,45 @@ public static void createStore( FileSystemAbstraction fileSystem, String storeDi
         }
     }
 
-    private NeoStores newNeoStores( PageCache pageCache, RecordFormats recordFormats )
+    private NeoStores newNeoStores()
+    {
+        return newNeoStores( storeDir );
+    }
+
+    private NeoStores newNeoStores( File storeDir )
     {
-        BatchingIdGeneratorFactory idGeneratorFactory = new BatchingIdGeneratorFactory( fileSystem );
-        StoreFactory storeFactory = new StoreFactory( storeDir, neo4jConfig, idGeneratorFactory, pageCache, fileSystem,
-                recordFormats, logProvider );
+        StoreFactory storeFactory = newStoreFactory( pageCache, recordFormats, storeDir );
         return storeFactory.openAllNeoStores( true );
     }
 
+    private StoreFactory newStoreFactory( PageCache pageCache, RecordFormats recordFormats, File storeDir )
+    {
+        StoreFactory storeFactory = new StoreFactory( storeDir, neo4jConfig,
+                new BatchingIdGeneratorFactory( fileSystem ), pageCache, fileSystem, recordFormats, logProvider );
+        return storeFactory;
+    }
+
+    /**
+     * The idea with replacement stores is that at some point a store is decided to be restructured
+     * one way or another, so that store is opened in the replacement store. When the {@link BatchingNeoStores}
+     * is closed those replacement stores will be swapped in.
+     */
+    public NeoStores getReplacementNeoStores( StoreType... stores )
```

Comment:
i found this concept extremely strange and don't think it should be batching neo stores responsibility to manage store replacements, this functionality should be extracted